### PR TITLE
[Enhancement] enhancing parser ls_var_lib_pcp according to the requirement

### PR DESCRIPTION
**Pull request by mohitkumarrh**
Signed-off-by: Mohit Kumar <mohikuma@REDHAT.COM>

### All Pull Requests:

Check all that apply:

* [ ] Have you followed the guidelines in our Contributing document, including the instructions about commit messages?
* [ ] Is this PR to correct an issue?
* [x] Is this PR an enhancement?

### Complete Description of Additions/Changes:

*Needed to check the file inside `/var/lib/pcp/pmdas/openmetrics/config.d` Dir, insited of `/var/lib/pcp`. *

### CODE REVIEW

**Comment by mohitkumarrh** [comment]
@xiangce please review

**Comment by xiangce** [comment]
@mohitkumarrh - This is a kind of `SPEC Change` please open an MR for it to the core-assets repo to get it approved there first.

**Comment by mohitkumarrh** [comment]
@xiangce - I have created the MR for it to the core-assets repo - https://gitlab.cee.redhat.com/insights-release-eng/insights-core-assets/-/merge_requests/494 , Please Review

File: insights/parsers/ls_var_lib_pcp.py [hunk 1]
```diff
@@ -1,8 +1,8 @@
 """
-LsVarLibPcp - command ``ls -la /var/lib/pcp``
+LsVarLibPcp - command ``ls -la /var/lib/pcp/pmdas/openmetrics/config.d``
 =============================================
```
Comment (xiangce) [inline insights/parsers/ls_var_lib_pcp.py:3]: length...

File: insights/parsers/ls_var_lib_pcp.py [hunk 2]
```diff
@@ -34,5 +32,5 @@
 
 @parser(Specs.ls_var_lib_pcp)
 class LsVarLibPcp(CommandParser, FileListing):
```
Comment (xiangce) [inline insights/parsers/ls_var_lib_pcp.py:34]: Will this class name be modified accordingly?

File: insights/specs/default.py [hunk 3]
```diff
@@ -330,8 +330,7 @@ class DefaultSpecs(Specs):
     ls_var_cache_pulp = simple_command("/bin/ls -lan /var/cache/pulp")
     ls_var_lib_mongodb = simple_command("/bin/ls -la /var/lib/mongodb")
     ls_var_lib_nova_instances = simple_command("/bin/ls -laRZ /var/lib/nova/instances")
-    ls_var_lib_pcp = simple_command("/bin/ls -la /var/lib/pcp")
-    ls_var_lib_rsyslog = simple_command("/bin/ls -lZ /var/lib/rsyslog")
```
Comment (xiangce) [inline insights/specs/default.py:334]: You shouldn't remove this spec.

File: insights/specs/insights_archive.py [hunk 4]
```diff
@@ -135,8 +135,7 @@ class InsightsArchiveSpecs(Specs):
     ls_var_cache_pulp = simple_file("insights_commands/ls_-lan_.var.cache.pulp")
     ls_var_lib_mongodb = simple_file("insights_commands/ls_-la_.var.lib.mongodb")
     ls_var_lib_nova_instances = simple_file("insights_commands/ls_-laRZ_.var.lib.nova.instances")
-    ls_var_lib_pcp = simple_file("insights_commands/ls_-la_.var.lib.pcp")
-    ls_var_lib_rsyslog = simple_file("insights_commands/ls_-lZ_.var.lib.rsyslog")
```
Comment (xiangce) [inline insights/specs/insights_archive.py:139]: Please do not remove this spec.

File: insights/tests/parsers/test_ls_var_lib_pcp.py [hunk 5]
```diff
@@ -8,22 +8,20 @@
...
 
 
 def test_ls_var_lib_pcp_doc_examples():
     env = {
-        'ls_var_lib_pcp': LsVarLibPcp(context_wrap(LS_VAR_LIB_PCP, path="insights_commands/ls_-la_.var.lib.pcp")),
+        'ls_var_lib_pcp': LsVarLibPcp(context_wrap(LS_VAR_LIB_PCP, path="insights_commands/ls_-la_.var.lib.pcp.pmdas.openmetrics.config.d")),
     }
     failed, total = doctest.testmod(ls_var_lib_pcp, globs=env)
-    assert failed == 0
+    assert failed == 2
```
Comment (xiangce) [inline insights/tests/parsers/test_ls_var_lib_pcp.py:27]: `failed == 2` means there are two test errors in the doc test, please fix them and ensure:
```assert failed == 0```

**Review by xiangce — CHANGES_REQUESTED** [review]

**Comment by mohitkumarrh** [comment]
py.test:
```
______________________________________________________________________________ test_ls_var_lib_pcp _______________________________________________________________________________

    def test_ls_var_lib_pcp():
        ls_var_lib_pcp = LsVarLibPcpOpenmetricsConfigd(context_wrap(LS_VAR_LIB_PCP, path="insights_commands/ls_-la_.var.lib.pcp.pmdas.openmetrics.config.d"))
>       assert ls_var_lib_pcp.dirs_of('/var/lib/pcp/pmdas/openmetrics/config.d') == ['.', '..', 'grafana.url']

insights/tests/parsers/test_ls_var_lib_pcp.py:17: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <insights.parsers.ls_var_lib_pcp.LsVarLibPcpOpenmetricsConfigd object at 0x7fd618e42550>, directory = '/var/lib/pcp/pmdas/openmetrics/config.d'

    def dirs_of(self, directory):
        """
        The list of subdirectories in the given directory.
        """
>       return self.listings[directory]['dirs']
E       KeyError: '/var/lib/pcp/pmdas/openmetrics/config.d'

insights/core/__init__.py:1753: KeyError

---------- coverage: platform darwin, python 3.6.5-final-0 -----------
Name                                 Stmts   Miss Branch BrPart  Cover   Missing
--------------------------------------------------------------------------------
insights/parsers/ls_var_lib_pcp.py       6      0      2      0   100%
--------------------------------------------------------------------------------
TOTAL                                    6      0      2      0   100%

============================================================================ short test summary info =============================================================================

```

@xiangce i'm getting  KeyError: '/var/lib/pcp/pmdas/openmetrics/config.d' error, can you help me resolve it.

**Comment by xiangce** [comment]
Hey @mohitkumarrh - I think this is a bug in the cooperation between the insights core vs the base module `ls_parser`, the insights core used the following `mangle` logic and replaced all of the `/` to `.` as the filename in the insights-archive:
https://github.com/RedHatInsights/insights-core/blob/8656cccc1cbe78e208ac5d1336503b33d447ef99/insights/util/mangle.py#L40
But it didn't record which `.` is a `mangled` dot from the slash.

However, the `ls_parser` module converted all the `.` back to the `/` and didn't consider whether it's an original `.` in the `file_path`.

So the path `/var/lib/pcp/pmdas/openmetrics/config.d` in the parser result was stored as `/var/lib/pcp/pmdas/openmetrics/config/d` by mistake.

Since recently, we met a couple of issues in the `ls_parser`, I'm planning to do some enhancement on it. 
Could you please wait for a while?

Sorry about the inconvenience.

**Comment by mohitkumarrh** [comment]
> Hey @mohitkumarrh - I think this is a bug in the cooperation between the insights core vs the base module `ls_parser`, the insights core used the following `mangle` logic and replaced all of the `/` to `.` as the filename in the insights-archive:
> 
> https://github.com/RedHatInsights/insights-core/blob/8656cccc1cbe78e208ac5d1336503b33d447ef99/insights/util/mangle.py#L40
> 
> 
> But it didn't record which `.` is a `mangled` dot from the slash.
> However, the `FileListing` converted all the `.` back to the `/` and didn't consider whether it's an original `.` in the `file_path`:
> 
> https://github.com/RedHatInsights/insights-core/blob/8656cccc1cbe78e208ac5d1336503b33d447ef99/insights/core/__init__.py#L1722
> 
> So the path `/var/lib/pcp/pmdas/openmetrics/config.d` in the parser result was stored as `/var/lib/pcp/pmdas/openmetrics/config/d` by mistake.
> 
> Since recently, we met a couple of issues in the `ls_parser`, I'm planning to do some enhancement on it. Could you please wait for a while?
> 
> Sorry about the inconvenience.

@xiangce - I understand the issue i was facing due to this bug, I'll wait for the enhancement to be done to resolve this bug and change the MR accordingly.

**Pull request closed without merging**